### PR TITLE
Allow topics to restrict editing to only the author or admins

### PR DESCRIPTION
**Pull request by Will-Howard**
From the ticket:
> Context: Pablo wants to make a EA version of the Stanford Encyclopedia of Philosophy. He would pay high profile names to write articles. In order to have those high profile names be happy with contributing, he wants to turn off collaborating on the wiki article.

I have added a flag `authorOnly` which can be set by the author when creating the topic, and only later changed by admins

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1202766096079144) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/tagging/TagPageButtonRow.tsx
Comment (Will-Howard) [inline packages/lesswrong/components/tagging/TagPageButtonRow.tsx:57]: this wasn't used so I repurposed it

File: packages/lesswrong/components/tagging/TagPageButtonRow.tsx
Comment (Will-Howard) [inline packages/lesswrong/components/tagging/TagPageButtonRow.tsx:116]: unrelated but think this will be helpful as I couldn't work out why the edit button wasn't appearing to begin with

File: packages/lesswrong/components/tagging/TagPageButtonRow.tsx
Comment (Will-Howard) [inline packages/lesswrong/components/tagging/TagPageButtonRow.tsx:179]: I did autoformat this file... but the diff is not too bad

File: packages/lesswrong/components/tagging/TagPageButtonRow.tsx
Comment (jpaddison3) [inline packages/lesswrong/components/tagging/TagPageButtonRow.tsx:104]: Currently this returns a falsey value if not logged in, which probably isn't what you wanted.

File: packages/lesswrong/components/tagging/TagPageButtonRow.tsx
Comment (jpaddison3) [inline packages/lesswrong/components/tagging/TagPageButtonRow.tsx:144]: The solidness of the lock icon makes it appear a deeper grey than the other icons or text. You can lighten it till it optically looks right.

File: packages/lesswrong/components/tagging/TagPageButtonRow.tsx
Comment (jpaddison3) [inline packages/lesswrong/components/tagging/TagPageButtonRow.tsx:114]: This part (thanks for adding it!) shouldn't appear if it's author-locked

File: packages/lesswrong/components/tagging/TagPageButtonRow.tsx
Comment (jpaddison3) [reply packages/lesswrong/components/tagging/TagPageButtonRow.tsx:104]: Separately: if we greenlight an author, it shouldn't matter if they have 0 karma, they should be able to edit. This will matter, as I'm guessing some of Pablo's academics won't have Forum accounts (although who knows if he'll have them edit the Forum directly, still, we can build this "right" anyway.)

File: packages/lesswrong/lib/collections/tags/schema.ts
Comment (jpaddison3) [reply packages/lesswrong/lib/collections/tags/schema.ts:175]: I do not want any old tag getting this. This is only going to be allowed for Pablo's project.

**Comment by jpaddison3** [comment]
I don't think we should use userId for the author id. Reasonable for you to assume! But I'd like to be able to change an existing topic to have a new author, have multiple co-authors, allow admins to select the author using a UserSelect, and do so on topic creation.

File: packages/lesswrong/lib/collections/tags/schema.ts
Comment (jpaddison3) [inline packages/lesswrong/lib/collections/tags/schema.ts:171]: Instead of authorOnly, you could have "restrict to these authors"

File: packages/lesswrong/components/tagging/TagPageButtonRow.tsx
Comment (jpaddison3) [inline packages/lesswrong/components/tagging/TagPageButtonRow.tsx:110]:
```suggestion
      This article can only be edited by the author, please comment in the discussion to suggest changes
```

**Review by jpaddison3 — CHANGES_REQUESTED** [review]
Nice work. I left some substantive comments, but I still like the attempt.

File: packages/lesswrong/lib/collections/tags/helpers.ts
Comment (Will-Howard) [inline packages/lesswrong/lib/collections/tags/helpers.ts:43]: I added this function always allow admins to edit

File: packages/lesswrong/lib/collections/tags/schema.ts
Comment (Will-Howard) [inline packages/lesswrong/lib/collections/tags/schema.ts:171]: @jpaddison3 I've changed this to allow an array of users like you suggested

File: packages/lesswrong/components/tagging/TagPageButtonRow.tsx
Comment (Will-Howard) [reply packages/lesswrong/components/tagging/TagPageButtonRow.tsx]: This is the new lighter version:
Screenshot 2022-08-09 at 16 27 54

File: packages/lesswrong/components/tagging/TagPageButtonRow.tsx
Comment (jimrandomh) [reply packages/lesswrong/components/tagging/TagPageButtonRow.tsx:182]: I don't think small formatting errors block merging this PR, but if this file is in fact the result of an autoformatter, then that formatter is either crap or is not configured correctly for our codebase.

A few formatting problems I spotted: on lines 112-114 has `<div>` and `</div>` on different indent levels. 146-149 have open and close parens on different indent levels. Line 80 adds spaces around the `|` which we never do.

File: packages/lesswrong/lib/collections/tags/fragments.ts
Comment (jpaddison3) [inline packages/lesswrong/lib/collections/tags/fragments.ts:6]: Can now remove?

File: packages/lesswrong/components/tagging/TagPageButtonRow.tsx
Comment (jpaddison3) [reply packages/lesswrong/components/tagging/TagPageButtonRow.tsx:182]: I think Will is talking about an initial save before he noticed that his autoformatter was turned on, after which he turned it off.

**Review by jpaddison3 — APPROVED** [review]
Nice work, this will be useful, thanks

File: packages/lesswrong/lib/collections/tags/fragments.ts
Comment (Will-Howard) [reply packages/lesswrong/lib/collections/tags/fragments.ts:6]: ah yeah we can, I'll make a note to do this in one of my other PRs though

File: packages/lesswrong/components/tagging/TagPageButtonRow.tsx
Comment (Will-Howard) [reply packages/lesswrong/components/tagging/TagPageButtonRow.tsx:182]: yeah I don't have eslint set up it's just the "Typescript language features" from vscode

File: packages/lesswrong/lib/collections/tags/fragments.ts
Comment (jpaddison3) [reply packages/lesswrong/lib/collections/tags/fragments.ts:6]: Do try to instill a habit of reading the comments before merging, even on approval.

**Pull request merged**
